app/routes/index.js

```diff
@@ -45,12 +55,20 @@ router.get('/item/:id', async (req, res) => {
   }
 })
 
-router.get('/item/:id/metadata', async (req, res) => {
-  const id = req.params && parseInt(req.params.id, 10)
+const getMetadataResponse = async (id, metadataKey, res) => {
   if (Number.isInteger(id) && id !== 0) {
-    const { metadata: hash, id: getId } = await getItem(id)
+    const { metadata, id: getId } = await getItem(id)
     if (getId === id) {
       try {
+        const buffer = Buffer.alloc(METADATA_KEY_LENGTH) // metadata keys are fixed length
+        const metadataKeyBuf = Buffer.from(metadataKey)
+        metadataKeyBuf.copy(buffer, 0)
```

Comment:
This is ok, but I think you can avoid allocating the second buffer with a `buffer.write` call?

Reply:
yes changed, much better